### DISCUSSION TITLE
Use with DataRow or DataReader

### DISCUSSION THREAD

**Discussion by Cirkadia** [axuno/SmartFormat#355] [Q&A] [answered]
Hi,
Been using SmartFormat with objects nicely. But can I use with a DataRow? Don't seem to see any examples or syntax.
I 've been converting row by row to a Dictionary, but is there a more direct approach that I'm missing?
Thanks!
Zef

**Comment by axunonb** [comment, marked as answer]
`DataRow` is not a built-in `ISource`.
Some years ago I came across the same requirement, and actually it's a "one-liner" (probably the same as you found, and I'm not sure whether any better exists):
```C#
// convert DataRow to Dictionary<string, object?>
if (dataItem is DataRow row)
{
    dataItem = row.Table.Columns.Cast<DataColumn>().ToDictionary(c => c.ColumnName, c => row[c]);
}
```
